src/modules/AddOnProduct.rb

```diff
@@ -145,7 +145,9 @@ def main
         "sle-hae"       => [ "sle-ha" ],
         "SUSE_SLES_SAP" => [ "SLES_SAP" ],
         # SMT is now integrated into the base SLES
-        "sle-smt"       => [ "SLES" ]
+        "sle-smt"       => [ "SLES" ],
+        # SLES_SAP includes SLES. So SLES is not needed anymore
+        "SLES"          => [ "SLES_SAP" ]
```

Comment:
OK, after thinking a bit more about it it turns out that it should actually work. The code should not preselect any conflicting products in first place.

And evaluating the products dependencies would be a bit tricky, e.g. you have product A and product B which conflicts with A. Which one should we select? A or B?

That means we cannot provide a generic solution which would work in all cases... :frowning_face: 
We might design it so it works in the SLES4SAP case, but it might not work in other cases. And the result is the same as if would use a simple hardcoded solution.

Anyway, I still think that for this purpose we should create a separate mapping, i.e. we have `@product_renames` so create `@product_conflicts` and use it for selecting products.

This would be a hardcoded list, which I do not like :frowning_face:, but I guess we cannot provide a better and still simple generic solution for all cases...

The rationale is that while `@product_renames` contains the very same products where only with  internal ID has been changed, for the conflicting products it is a different case. These products are really different, there is no connection between them.